common/src/main/java/org/mvndaemon/mvnd/common/Environment.java

```diff
@@ -94,9 +94,16 @@ public enum Environment {
     // Maven properties
     //
     /** The path to the Maven local repository */
-    MAVEN_REPO_LOCAL("maven.repo.local", null, null, OptionType.PATH, Flags.NONE),
+    MAVEN_REPO_LOCAL("maven.repo.local", null, null, OptionType.PATH, Flags.DISCRIMINATING | Flags.OPTIONAL),
     /** The location of the maven settings file */
-    MAVEN_SETTINGS("maven.settings", null, null, OptionType.PATH, Flags.NONE, "mvn:-s", "mvn:--settings"),
+    MAVEN_SETTINGS(
+            "maven.settings",
+            null,
+            null,
+            OptionType.PATH,
+            Flags.DISCRIMINATING | Flags.OPTIONAL,
+            "mvn:-s",
+            "mvn:--settings"),
```

Comment:
@oehme Should this also apply to `-gs` and `--global-settings`?

Reply:
Yes, it should, but from a cursory look at this class it's not yet supported at all. I also think that the Java Home should be discriminating and that the actual content of the settings.xml should be considered as well, but all of these seemed out of scope for this PR.